apisix/plugins/hmac-auth.lua

```diff
@@ -244,7 +244,13 @@ local function generate_signature(ctx, secret_key, params)
 
             -- whether to encode the uri parameters
             if type(param) == "table" then
+                local vals = {}
```

Comment:
i  wonder that  why not use table.new

Reply:
Just keep the same code style like table `keys`

Reply:
Not necessary to keep old code style here. We can specify the initialization size when we create a new table by using `core.table.new(narray, nhash)`